### PR TITLE
Update SW & HW page

### PR DESCRIPTION
**Pull request by tyhoff**
Some simple changes have been made:
- Recommend more strongly SemVer 2.0 (and fix all naming of this)
- Move natsort & testing section down to the bottom where it belonds to keep more important topics towards the top/middle of the page
- Remove generic information from the MCU-specific sections, and vice versa
- Be more consistent about positioning of the Requirements section
- Say why a sane versioning scheme is important
- Only provide one way to test natsort, instead of providing multiple ways.
- Grammarly the whole page

### CODE REVIEW

**Comment by tyhoff** [comment]
I think if we take in Noah's OTA post, we'll want to take the Software Version best practices & the  Software + Release naming best practice and put it into this doc and the OTA one. I don't think we want to duplicate that data.

File: docs/platform/software-version-hardware-version.mdx [hunk 1]
```diff
@@ -148,72 +144,33 @@ to:
   firmware, the firmware's Software Version is used to determine if a device
   requires an update.
 
-### Version Schemes {#version-schemes}
+### Versioning Schemes {#version-schemes}
+
+Human-readable and understandable software versions go a long way. It allows
+everyone involved in a project to quickly determine what build is running on a
+device, whether it is new or old, or if it is a a one-off build built by an
+engineer. On top of readability, a good versioning scheme allows trivial and
```
Comment (ejohnso49) [inline docs/platform/software-version-hardware-version.mdx:152]:
```suggestion
Human-readable and understandable software versions go a long way. It allows
everyone involved in a project to quickly determine whether a device's build is new, old, or a one-off build built by an
engineer. On top of readability, a good versioning scheme allows trivial and
```

**Review by ejohnso49 — APPROVED** [review]
Great refresh

**Review by vlindhol — APPROVED** [review]

File: docs/platform/software-version-hardware-version.mdx [hunk 2]
```diff
@@ -12,21 +12,17 @@ can be used to identify different device types being tracked in the same Project
 (i.e. `smart-toaster` and `smart-oven`) and different board revisions of the
 same device type (i.e. `smart-toaster-evt`, `smart-toaster-pvt`, etc.).
 
+By default, the Hardware Version of an individual Device is set during the first
```
Comment (noahp) [inline docs/platform/software-version-hardware-version.mdx:15]:
```suggestion
By default, the Hardware Version of an individual Device is set permanently during the first
```

File: docs/platform/software-version-hardware-version.mdx [hunk 3]
```diff
@@ -431,3 +325,87 @@ A Software Type/Version cannot be unarchived so proceed with caution!
...
+1.0.0+ffffffff
+```
+
+### Testing the Ordering {#testing}
+
+It is imperative to understand how your Software Versions will be ordered by
+Memfault's OTA system.
+
+To test version ordering locally, install
+[Natsort](https://pypi.org/project/natsort/) and create a simple Python script
```
Comment (noahp) [inline docs/platform/software-version-hardware-version.mdx:394]: I greatly prefer the example using the `natsort` command line utility vs. creating a script like this; IMO the intersection of users needing this, and the relative confusion/complexity between the 2 variants, makes it OK (and simpler).

**Review by noahp — APPROVED** [review]
Sweet! tyty

**Pull request merged**